executor/executor.go

```diff
@@ -674,8 +674,21 @@ func (e *ShowDDLJobQueriesExec) Open(ctx context.Context) error {
 		return err
 	}
 
-	e.jobs = append(e.jobs, jobs...)
-	e.jobs = append(e.jobs, historyJobs...)
+	appendedJobID := make(map[int64]struct{})
+	// deduplicate job results
+	// for situations when this operation happens at the same time with new DDLs being executed
```

Comment:
the reason of duplication I think is jobs may finish between `ddl.GetAllDDLJobs` and `ddl.GetLastNHistoryDDLJobs`, so we will see it in both pending DDL and finished DDL. No need to check duplication for `jobs` from `ddl.GetAllDDLJobs`

Reply:
I tried to not check duplication for `jobs` from `ddl.GetAllDDLJobs`, but sometimes may result in this error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6130472]

goroutine 15610 [running]:
github.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1()
	/Users/apple/Desktop/go/tidb/executor/adapter.go:440 +0x3b3
panic({0x6f5fea0, 0xa347510})
	/usr/local/go/src/runtime/panic.go:884 +0x212
github.com/pingcap/tidb/planner/core.NormalizeFlatPlan(0xc004400798?)
	/Users/apple/Desktop/go/tidb/planner/core/encode.go:265 +0x52
github.com/pingcap/tidb/executor.getPlanDigest(0xc004400798)
	/Users/apple/Desktop/go/tidb/executor/adapter.go:1651 +0x53
github.com/pingcap/tidb/executor.(*ExecStmt).observeStmtBeginForTopSQL(0xc0071733b0, {0x7dc7390, 0xc007742270})
	/Users/apple/Desktop/go/tidb/executor/adapter.go:1840 +0x90
github.com/pingcap/tidb/executor.(*ExecStmt).Exec(0xc0071733b0, {0x7dc7390, 0xc007742270})
	/Users/apple/Desktop/go/tidb/executor/adapter.go:505 +0x705
github.com/pingcap/tidb/session.runStmt({0x7dc7320, 0xc00019c008}, 0xc000804c80, {0x7dd9300, 0xc0071733b0?})
	/Users/apple/Desktop/go/tidb/session/session.go:2360 +0x565
github.com/pingcap/tidb/session.(*session).ExecuteStmt(0xc000804c80, {0x7dc7320, 0xc00019c008}, {0x7ddf450?, 0xc0071732c0})
	/Users/apple/Desktop/go/tidb/session/session.go:2224 +0xaed
github.com/pingcap/tidb/testkit.(*TestKit).ExecWithContext(0xc0042f14a0, {0x7dc7320, 0xc00019c008}, {0x74f5c14, 0x1d}, {0x0, 0x0, 0x0})
	/Users/apple/Desktop/go/tidb/testkit/testkit.go:321 +0x71a
github.com/pingcap/tidb/testkit.(*TestKit).MustQueryWithContext(0xc0042f14a0, {0x7dc7320, 0xc00019c008}, {0x74f5c14, 0x1d}, {0x0, 0x0, 0x0})
	/Users/apple/Desktop/go/tidb/testkit/testkit.go:153 +0x145
github.com/pingcap/tidb/testkit.(*TestKit).MustQuery(0xc005e5dce0?, {0x74f5c14?, 0xc000a96628?}, {0x0?, 0x5ef9dd5?, 0x7dafbd0?})
	/Users/apple/Desktop/go/tidb/testkit/testkit.go:147 +0x8d
github.com/pingcap/tidb/executor_test.TestAdmin.func3()
	/Users/apple/Desktop/go/tidb/executor/executor_test.go:5650 +0xd1
created by github.com/pingcap/tidb/executor_test.TestAdmin
	/Users/apple/Desktop/go/tidb/executor/executor_test.go:5646 +0x201b


Process finished with the exit code 1
```

I guess there may still exist duplication for `jobs` from `ddl.GetAllDDLJobs` for some reason?